### ISSUE TITLE
ItemList truncation does not show ellipsis (...) with new script editor list minimum width

### ISSUE THREAD

**Issue by tetrapod00** [godotengine/godot#97124]
### Tested versions

4.3 (for MRP)
4.4 (exposed to the actual editor since https://github.com/godotengine/godot/pull/96909)

### System information

Windows 10

### Issue description

In an ItemList, names are truncated if they are too wide for the list. Usually this is done with an ellipsis `...`, but when the width is small enough, no ellipsis is used. I think this is good intended behavior, otherwise only an ellipsis would be shown at very low widths.

Since https://github.com/godotengine/godot/pull/96909 set the minimum width of the script list to 100, the width is small enough that many script names are now truncated with no ellipsis. This happens *right around* a script list width of 100. For `new_script.gd`, the threshold for the behavior happens at 101px wide, so it was easily noticeable when testing.

Observed behavior in the https://github.com/godotengine/godot/pull/96909 artifact:

https://github.com/user-attachments/assets/17179298-3a6d-4d83-b0af-7be34a7d144e

Observed behavior in the MRP. 

https://github.com/user-attachments/assets/bd7a7570-f944-4e00-a27c-fe37b4597373

Possible resolutions:
- Tweak the truncation limits so that an ellipsis is still used at the minimum script width of 100.
- Tweak the minimum script width to 105 or 110, to barely avoid this behavior.
- Do nothing, as this *is* intended behavior, just an edge case that usually doesn't come up in the editor. In this case the user is knowingly making their editor UI crowded in exchange for extra script editor space, and minor "problems" like this may be exppected. (It's me. I'm user. 😃)

CC @aXu-AP

### Steps to reproduce

Open the MRP. 
Adjust the HSplitContainer left and right. You can test with and without 16x16 icons.

### Minimal reproduction project (MRP)

[mrp-truncation.zip](https://github.com/user-attachments/files/17032862/mrp-truncation.zip)

**Comment by AThousandShips** [comment]
> 4.3 (for MRP)

Don't list versions that *aren't* affected, unless you point out that they aren't affected, unless this isn't a regression from #96909 this isn't in 4.3

**Comment by AThousandShips** [comment]
So to clarify, this is a pre-existing issue in `ItemList` that affects them already? But this report is specifically about how it affects the editor?

I think this should be solved in `ItemList` and a report should focus on the behaviour there and deeper details to test it, rather than trying to fix one symptom of it

**Comment by tetrapod00** [comment]
Yes, this is a pre-existing behavior of ItemList (and maybe deeper code that actually does the truncation). It is good that at some point, item list labels are truncated without ellipses. As an example, `new_script.gd` truncated to 5 characters is `new_s` rather than `ne...`.

There is some threshold for when this happens, set somewhere. Before https://github.com/godotengine/godot/pull/96909, the script editor list only showed truncations with ellipses, because it was sufficiently wide. So this behavior was not exposed to the editor, at least at that location.

I don't know how much thought went into the current truncation threshold is for ItemList text. I also don't know all the places where it could be used.

I opened this issue specifically focusing on the script editor list because it came out of a [comment ](https://github.com/godotengine/godot/pull/96909#issuecomment-2346879290) from https://github.com/godotengine/godot/pull/96909. It's possible that either a local fix to the script editor *or* a deeper change to ItemList or truncation behavior is the right solution.

**Comment by AThousandShips** [comment]
I think fixing this in `ItemList` is far more critical, I don't think it's very critical that it now doesn't work in the default state in the script editor, compared to that, so I'd say to open a dedicated issue report for the `ItemList` behavior and we'll see if we close this one or keep both open

**Comment by tetrapod00** [comment]
I think that issue would look very close to this issue, just with a different title and slightly different emphasis? The MRP I posted for this one will work for testing the underlying issue.

**Comment by AThousandShips** [comment]
It wouldn't, this issue is about the script editor, not `ItemList`, so either this should be changed to focus on `ItemList` or a different issue should be opened, we can't track two very different scopes of issue in the same issue report, that is a recipe for making bad errors

This issue would be solved by *either* changing the default size in the editor, *or* fixing the issue in `ItemList`, but the `ItemList` issue can't be solved by changing the editor code

It's really important to focus on the right place

**Comment by tetrapod00** [comment]
Done in https://github.com/godotengine/godot/issues/97125.

**Comment by WhalesState** [comment]
It's related to `TextServer`, this issue exists with all the nodes that can change `text_overrun_behavior`.
The ellipsis only appears if there's enough space for it's text after at least 5 characters, it's very old issue, and maybe can be reproduced on 4.0 or 4.1 too.
